### PR TITLE
update katib laucher

### PR DESCRIPTION
**Pull request by hougangliu**
1. sync up since katib directory org changed
2. add option deleteafterdone

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1118)

### CODE REVIEW

File: components/kubeflow/katib-launcher/component.yaml [hunk 1]
```diff
@@ -5,15 +5,16 @@ inputs:
...
 - {name: Metrics names,             type: String,                       description: 'List of metric names (comma-delimited).'}
 - {name: Parameter configs,         type: YAML,     default: '',        description: 'Parameter configs (YAML/JSON format).'}
 - {name: NAS config,                type: YAML,     default: '',        description: 'NAS config (YAML/JSON format).'}
 - {name: Worker template path,      type: String,   default: '',        description: 'Worker spec.'}
 - {name: Metrics collector template path, type: String, default: '',    description: 'Metrics collector spec.'}
 - {name: Suggestion spec,           type: YAML,     default: '',        description: 'Suggestion spec (YAML/JSON format).'}
-- {name: StudyJob timeout minutes,  type: Integer,  default: '10',      description: 'Time in minutes to wait for the StudyJob to complete'}
+- {name: StudyJob timeout minutes,  type: Integer,  default: '10',      description: 'Time in minutes to wait for the StudyJob to complete.'}
+- {name: Delete StudyJob flag,      type: String,   default: 'True',    description: 'When StudyJob done, delete the StudyJob if it is True.'}
```
Comment (Ark-kun) [inline components/kubeflow/katib-launcher/component.yaml]:
```suggestion
- {name: Delete finished job,       type: Boolean,  default: 'True',    description: 'Whether to delete the job after it's finished.'}
```

File: components/kubeflow/katib-launcher/component.yaml [hunk 2]
```diff
@@ -34,5 +35,6 @@ implementation:
       --mcollectortemplatepath, {inputValue: Metrics collector template path},
       --suggestionspec,         {inputValue: Suggestion spec},
       --studyjobtimeoutminutes, {inputValue: StudyJob timeout minutes},
+      --deleteafterdone,        {inputValue: Delete StudyJob flag},
```
Comment (Ark-kun) [inline components/kubeflow/katib-launcher/component.yaml]:
```suggestion
      --deleteafterdone,        {inputValue: Delete finished job},
```

File: components/kubeflow/katib-launcher/kubeflow_katib_launcher_op.py [hunk 3]
```diff
@@ -34,6 +34,7 @@ def kubeflow_studyjob_launcher_op(name, namespace, optimizationtype, objectiveva
             "--mcollectortemplatepath", mcollectortemplatepath,
             "--suggestionspec", suggestionspec,
             "--outputfile", output_file,
+            "--deleteafterdone", delete,
```
Comment (Ark-kun) [inline components/kubeflow/katib-launcher/kubeflow_katib_launcher_op.py]: `delete_finished` ?

File: components/kubeflow/katib-launcher/src/launch_study_job.py [hunk 4]
```diff
@@ -39,6 +39,14 @@ def yamlOrJsonStr(str):
 def strToList(str):
     return str.split(",")
 
+def str2bool(v):
```
Comment (Ark-kun) [inline components/kubeflow/katib-launcher/src/launch_study_job.py]: `from distutils.util import strtobool`

File: components/kubeflow/katib-launcher/src/launch_study_job.py [hunk 5]
```diff
@@ -121,6 +129,9 @@ def main(argv=None):
   parser.add_argument('--outputfile', type=str,
                       default='/output.txt',
                       help='The file which stores the best trial of the studyJob.')
+  parser.add_argument('--deleteafterdone', type=str2bool,
```
Comment (Ark-kun) [inline components/kubeflow/katib-launcher/src/launch_study_job.py]: `--delete-after-done` ?

**Comment by hougangliu** [comment]
@Ark-kun thanks for your comment. Updated

**Comment by Ark-kun** [comment]
> @Ark-kun thanks for your comment. Updated

Can you please quote 'True' and improve the description: https://github.com/kubeflow/pipelines/pull/1118#discussion_r273654131

**Comment by hougangliu** [comment]
> Can you please quote 'True' and improve the description: [#1118 (comment)](https://github.com/kubeflow/pipelines/pull/1118#discussion_r273654131)

sorry, miss the comment. Updated now. Thanks

**Pull request merged**